File: LAB 1(8-1-19)/comp.java
public class comp
{
	float real,img;
	public comp(float a,float b)
	{
		this.real=a;
		this.img=b;
	}
	public static void show(comp c)
	{
		if(c.img<0)
		{
			System.out.printf("%.1f%.1fi\n",c.real,c.img);
		}
		else
		{
			System.out.printf("%.1f+%.1fi\n",c.real,c.img);
		}
		
	}
	public static comp add(comp c1,comp c2)
	{
		comp c=new comp(0,0);
		c.real=c1.real+c2.real;
		c.img=c1.img+c2.img;
		return(c);
	}
	public static comp sub(comp c1,comp c2)
	{
		comp c=new comp(0,0);
		c.real=c1.real-c2.real;
		c.img=c1.img-c2.img;
		return(c);
	}
	public static comp mul(comp c1,comp c2)
	{
		comp c=new comp(0,0);
		c.real=(c1.real*c2.real)-(c1.img*c2.img);
		c.img=(c1.img*c2.real)+(c1.real*c2.img);
		return(c);
	}
	public static comp div(comp c1,comp c2)
	{
		comp c=new comp(0,0);
		float z=(c2.real*c2.real)+(c2.img*c2.img);
		c.real=((c1.real*c2.real)+(c1.img*c2.img))/z;
		c.img=((c1.img*c2.real)-(c1.real*c2.img))/z;
		return(c);
	}
	public static void main(String[] args)
	{
			comp c1=new comp(3,4);
			comp c2=new comp(2,5);
			comp c=add(c1,c2);
			System.out.printf("addition is::");
			show(c);
			c=sub(c1,c2);
			System.out.printf("subtraction is::");
			show(c);
			c=mul(c1,c2);
			System.out.printf("multiplication is::");
			show(c);
			c=div(c1,c2);
			System.out.printf("division is::");
			show(c);
			
	}
}

File: LAB 1(8-1-19)/comp.cpp
#include<bits/stdc++.h>
using namespace std;
class comp
{
    float real,img;
    public:
	void input()
	{
		float a,b;
		cout<<"enter real part::";
		cin>>a;
		real=a;
		cout<<"enter imaginary part::";
		cin>>b;
		img=b;
	}
	void show()
	{
		if(img<0)
		{
			cout<<real<<"-i"<<-1*img<<endl;
		}
		else
		{
			cout<<real<<"+i"<<img<<endl;
		}
	}
	comp operator +(comp c)
	{
		comp ans;
		ans.real=real+c.real;
		ans.img=img+c.img;
		return ans;
	}
	comp operator -(comp c)
	{
		comp ans;
		ans.real=real-c.real;
		ans.img=img-c.img;
		return ans;
	}
	comp operator *(comp c)
	{
		comp ans;
		ans.real=(real*c.real)-(img*c.img);
		ans.img=(real*c.img)+(img*c.real);
		return ans;
	}
	comp operator /(comp c)
	{
		comp ans;
		float z=(c.real*c.real)+(c.img*c.img);
		float x=((real*c.real)+(img*c.img))/z;
		float y=((img*c.real)-(real*c.img))/z;
		ans.real=x;
		ans.img=y;
		return ans;
	}
};
int main()
{
	comp c1,c2;
	c1.input();
	c1.show();
	c2.input();
	c2.show();
	comp x;
	cout<<"choose options::"<<endl<<"1.add"<<endl<<"2.sutract"<<endl<<"3.multiply"<<endl<<"4.division"<<endl<<"5.exit"<<endl;
	while(1)
	{
		
		int ch;
		cin>>ch;
		if(ch==5)
		{
			break;
		}
		if(ch==1)
		{
			x=c1+c2;
			cout<<"addition is:";
			x.show();	
		}
		if(ch==2)
		{
			x=c1-c2;
			cout<<"subtraction is:";
			x.show();	
		}
		if(ch==3)
		{
			x=c1*c2;
			cout<<"multiplication is:";
			x.show();	
		}
		if(ch==4)
		{
			x=c1/c2;
			cout<<"division is:";
			x.show();	
		}
	}
	return 0;
}
